plugins/opensslcryptoplugin/evp/evp.c

```diff
@@ -7,6 +7,17 @@
 
 #include "evp_p.h"
 
+#define OSSLEVP_PRINT_ERR(message) \
+    fprintf(stderr, "%s#%d, %s: %s\n", __FILE__, __LINE__, __FUNCTION__, message);
```

Comment:
That is actually not new in this commit, that's how it was previously as well, ie. we print possible errors to `stderr`. Feel free to chime in if you have a better idea.

Reply:
How about returning the error to the caller and letting the caller print the error message in the manner the caller finds appropiate? There are Qt logging facilities, glib etc. Plenty of ways to make debugging and error messages available on demand or return them to the caller if the caller is interested. But that's what I like, and it could be that some people like to see those printed unconditionally to `stderr`, why not. Feel free to ignore. I just thought it was an oversight but apparently it's not.

Reply:
If you look at the calling code (`OpenSslCryptoPlugin`), you can see that an error is actually returned to the caller. However, since this is a daemon which is mostly used through D-Bus, the client cannot be expected to deal with low-level error codes from OpenSSL.

I guess `stderr` is a cheap way for debugging such issues. However this discussion is not really related to this pull request, since `stderr` was already used before. So I would suggest to open an issue where you can detail your suggestion and possible alternatives and we can continue the discussion.

@chriadam What do you think? Should we stick to `stderr` or use something else?

Reply:
It's a good idea to try to return the "actual" error to the client for debugging purposes.  By which I mean, it's not really (IMO) useful to try to modify application behaviour at runtime depending on the error message, but it is useful for client applications to see what they might have done wrongly...

In short, I think it's probably a good idea, but very low priority.